### DISCUSSION TITLE
Renovate tries to update gradle Feature Variant capabilities

### DISCUSSION THREAD

**Discussion by patrickneulichedldmde** [renovatebot/renovate#33144] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Renovate 39.58.1 on GitLab using your renovate image

### Please tell us more about your question or problem

We use gradle Feature Variants and specify custom capabilities to make the mutually exclusive as shown [here](https://docs.gradle.org/current/userguide/feature_variants.html#ex-a-producer-of-multiple-features-that-are-mutually-exclusive).
Renovate recognizes this as a dependency and tries lookup `org.gradle.demo:producer-db-support` as a maven artifact but does not find it for obvious reasons.
It outputs a warning saying "Failed to look up maven package org.gradle.demo:producer-db-support".

It seem the gradle managers matching is quite generous since it also tries to lookup any three comma separated string where the last string has a number in it.
I out case it tries to find `team1:team2` as maven artifact for the following gradle code setOf("team1", "team2", "team3").

This brings us quite some headache. I'm not so fluent with typescript otherwise I'd try to create a PR.
So I kindly ask if someone could check if there is a good way to filter them out.
I think [here](https://github.com/renovatebot/renovate/blob/main/lib/modules/manager/gradle/extract.spec.ts#L69) would be a good point to start?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by Churro** [comment, marked as answer]
Fixed by https://github.com/renovatebot/renovate/pull/33438

**Comment by patrickneulichedldmde** [comment]
Thank you so much!
